src/App.js

```diff
@@ -2,19 +2,36 @@ import * as React from "react";
 import TodoList from "./TodoList";
 import AddTodoForm from "./AddTodoForm";
 
+// Custom hook for semi-persistent state
+function useSemiPersistentState(key, initialState) {
+  const [state, setState] = React.useState(
+    localStorage.getItem(key) || initialState
+  );
```

Comment:
Good catch on setting the initial state! 🥇 

You can also set it as an empty array and parse the value when retrieving it from localStorage:

```
function useSemiPersistentState(key) {
  const [state, setState] = React.useState(
    JSON.parse(localStorage.getItem(key)) || []
 );
 ```